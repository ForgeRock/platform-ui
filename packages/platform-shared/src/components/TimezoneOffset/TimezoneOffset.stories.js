/**
 * Copyright 2020 ForgeRock AS. All Rights Reserved
 *
 * Use of this code requires a commercial software license with ForgeRock AS.
 * or with one of its affiliates. All use shall be exclusively subject
 * to such license between the licensee and ForgeRock AS.
 */
/* eslint import/no-extraneous-dependencies: 0 */
import { storiesOf } from '@storybook/vue';
import { withKnobs, text } from '@storybook/addon-knobs';
import {
  BCard,
  BContainer,
} from 'bootstrap-vue';
import TimezoneOffset from './index';

const stories = storiesOf('Components|TimezoneOffset', module).addParameters({ component: TimezoneOffset });

stories.addDecorator(withKnobs);

const template = `
  <div class="margin-25">
    <div class="sg_masthead">
      <BContainer>
        <h1 class="display-4">TimezoneOffset</h1>
        <p class="mb-4 lead">
          Input that allows user to select a timezone offset from GMT. Uses Bootstrap input with type range. Formats display value to user friendly text.
        </p>
      </BContainer>
    </div>
    <BContainer>
      <section class="sg_pattern">
        <h2>Basic Example</h2>
        <p class="mb-5">
          Select a timezone offset.
        </p>
        <BCard>
          <FrTimezoneOffset 
            :placeholder="placeholder"
            v-model="value" />
          Value (measured in hours): {{ value }}
        </BCard>
      </section>
    </BContainer>
  </div>
`;

stories.add('TimezoneOffset', () => ({
  components: {
    FrTimezoneOffset: TimezoneOffset,
    BCard,
    BContainer,
  },
  template,
  data() {
    return {
      value: -7,
    };
  },
  props: {
    placeholder: {
      type: String,
      default: text('placeholder', 'Placeholder Text'),
    },
  },
}));
