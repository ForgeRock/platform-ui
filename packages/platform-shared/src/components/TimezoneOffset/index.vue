<!-- Copyright (c) 2020-2021 ForgeRock. All rights reserved.

This software may be modified and distributed under the terms
of the MIT license. See the LICENSE file for details. -->
<template>
  <div>
    <BDropdown
      class="timezone-dropdown w-100"
      menu-class="shadow-lg"
      no-caret
      toggle-class="p-0"
      variant="none"
      v-bind="$attrs">
      <template #button-content>
        <FrField
          v-bind="$attrs"
          v-model="timezone"
          class="timezone-field"
          :label="placeholder">
          <template #prepend>
            <BInputGroupPrepend>
              <div :class="[{'disabled': $attrs.disabled}, 'input-group-text inset']">
                <FrIcon
                  name="location_on"
                />
              </div>
            </BInputGroupPrepend>
          </template>
        </FrField>
      </template>
      <output class="timezone-output form-control form-control-sm text-center">
        {{ offset | offsetString }}
      </output>
      <div class="pt-3 pb-0">
        <div class="marker" />
        <BFormInput
          v-model="offset"
          type="range"
          min="-12"
          max="14"
          step="0.25" />
      </div>
      <div class="d-flex justify-content-between">
        <span>
          <small>-12</small>
        </span>
        <span>
          <small>+14</small>
        </span>
      </div>
    </BDropdown>
    <small class="form-text">
      {{ $t('timezone.description') }}
      <BLink
        href="https://www.timeanddate.com/time/zones/"
        target="_blank">
        {{ $t('timezone.linkText') }}
      </BLink>
    </small>
  </div>
</template>

<script>
import {
  BDropdown,
  BFormInput,
  BInputGroupPrepend,
  BLink,
} from 'bootstrap-vue';
import FrField from '@forgerock/platform-shared/src/components/Field';
import FrIcon from '@forgerock/platform-shared/src/components/Icon';

/**
 * Input that allows user to select a timezone offset. Uses BInput with type range.
 * Display values are filtered to a user friendly string, but v-model value is a number.
 * Example string: 'GMT +/- hh:mm'
 */
export default {
  name: 'TimezoneOffset',
  components: {
    BDropdown,
    BFormInput,
    BInputGroupPrepend,
    FrField,
    FrIcon,
    BLink,
  },
  props: {
    /**
     * @model Number representation of offset
     */
    value: {
      type: Number,
      default: 0,
    },
    /**
     * Text label. Floats when a value is selected.
     */
    placeholder: {
      type: String,
      default: '',
    },
  },
  data() {
    return {
      offset: this.value,
      timezone: this.$options.filters.offsetString(this.value),
    };
  },
  filters: {
    /**
     * Converts numerical offset in hours to string 'GMT +/- hh:mm'
     * @param {Number} value offset from GMT in hours
     * @returns {String} offset from GMT
     */
    offsetString(value) {
      const numberVal = parseFloat(value);
      const isPositive = numberVal >= 0;
      const hours = parseInt(Math.abs(numberVal), 10);
      let minutes = (Math.abs(numberVal) % 1) * 60;
      if (minutes < 10) {
        minutes = `0${minutes.toString()}`;
      }
      const timeString = isPositive ? `GMT + ${hours}:${minutes}` : `GMT - ${hours}:${minutes}`;
      return timeString;
    },
  },
  watch: {
    offset(newVal) {
      this.timezone = this.$options.filters.offsetString(newVal);
      /**
       * triggered whenever the offset is changed.
       * @property {Number} newVal new offset value
       */
      this.$emit('input', Number(newVal));
    },
    value(newVal) {
      this.offset = newVal;
    },
  },
};
</script>
<style lang="scss" scoped>
.timezone-dropdown {
  /deep/ .dropdown-menu.show {
    min-width: 300px;
    padding: 0.5rem;
  }
}

.timezone-field {
  /deep/ .form-label-group-input {
    margin-left: -10px;

    /deep/ input {
      border-left: none !important;
    }
  }
}

.input-group-prepend .disabled {
  background-color: $gray-100;
}
</style>
