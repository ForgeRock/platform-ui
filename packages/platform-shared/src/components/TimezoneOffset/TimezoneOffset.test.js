/**
 * Copyright (c) 2020-2023 ForgeRock. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */

import { shallowMount } from '@vue/test-utils';
import TimezoneOffset from './index';

describe('TimezoneOffset Component', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(TimezoneOffset, {
      global: {
        mocks: {
          $t(val) { return val; },
        },
        renderStubDefaultSlot: true,
      },
    });
  });

  it('display value is user friendly string', async () => {
    await wrapper.setProps({ value: -7 });
    expect(wrapper.find('output').text()).toBe('GMT - 7:00');
  });

  it('contains link to timezone offset explanation', () => {
    expect(wrapper.findComponent({ name: 'BLink' }).text()).toBe('timezone.linkText');
    expect(wrapper.findComponent({ name: 'BLink' }).attributes('href')).toBe('https://www.timeanddate.com/time/zones/');
  });
});
