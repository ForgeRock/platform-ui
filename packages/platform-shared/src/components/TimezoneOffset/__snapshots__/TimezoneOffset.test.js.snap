// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimezoneOffset Component display value is user friendly string: console-warn-snapshot 1`] = `
[
  [
    "[Vue warn]: (deprecation RENDER_FUNCTION) Vue 3's render function API has changed. You can opt-in to the new API with:

  configureCompat({ RENDER_FUNCTION: false })

  (This can also be done per-component via the "compatConfig" option.)
  Details: https://v3-migration.vuejs.org/breaking-changes/render-function-api.html",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation GLOBAL_EXTEND) (2)",
    "
",
    " at <TimezoneOffset",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation COMPONENT_V_MODEL) v-model usage on component has changed in Vue 3. Component that expects to work with v-model should now use the "modelValue" prop and emit the "update:modelValue" event. You can update the usage and then opt-in to Vue 3 behavior on a per-component basis with \`compatConfig: { COMPONENT_V_MODEL: false }\`.
  Details: https://v3-migration.vuejs.org/breaking-changes/v-model.html",
    "
",
    " at <BFormInput",
    "modelValue=0",
    "onUpdate:modelValue=fn",
    "type="range"",
    " ...",
    ">",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "autofocus" with v-bind value \`false\` will render autofocus="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <BFormInput",
    "type="range"",
    "min="-12"",
    "max="14"",
    " ...",
    ">",
    "
",
    " at <BDropdown",
    "class="timezone-dropdown w-100"",
    "menu-class="shadow-lg"",
    "no-caret=""",
    " ...",
    ">",
    "
",
    " at <TimezoneOffset",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "disabled" with v-bind value \`false\` will render disabled="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <BFormInput",
    "type="range"",
    "min="-12"",
    "max="14"",
    " ...",
    ">",
    "
",
    " at <BDropdown",
    "class="timezone-dropdown w-100"",
    "menu-class="shadow-lg"",
    "no-caret=""",
    " ...",
    ">",
    "
",
    " at <TimezoneOffset",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "required" with v-bind value \`false\` will render required="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <BFormInput",
    "type="range"",
    "min="-12"",
    "max="14"",
    " ...",
    ">",
    "
",
    " at <BDropdown",
    "class="timezone-dropdown w-100"",
    "menu-class="shadow-lg"",
    "no-caret=""",
    " ...",
    ">",
    "
",
    " at <TimezoneOffset",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "ariaInvalid" with v-bind value \`false\` will render ariaInvalid="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <BFormInput",
    "type="range"",
    "min="-12"",
    "max="14"",
    " ...",
    ">",
    "
",
    " at <BDropdown",
    "class="timezone-dropdown w-100"",
    "menu-class="shadow-lg"",
    "no-caret=""",
    " ...",
    ">",
    "
",
    " at <TimezoneOffset",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "lazy" with v-bind value \`false\` will render lazy="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <BFormInput",
    "type="range"",
    "min="-12"",
    "max="14"",
    " ...",
    ">",
    "
",
    " at <BDropdown",
    "class="timezone-dropdown w-100"",
    "menu-class="shadow-lg"",
    "no-caret=""",
    " ...",
    ">",
    "
",
    " at <TimezoneOffset",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "lazyFormatter" with v-bind value \`false\` will render lazyFormatter="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <BFormInput",
    "type="range"",
    "min="-12"",
    "max="14"",
    " ...",
    ">",
    "
",
    " at <BDropdown",
    "class="timezone-dropdown w-100"",
    "menu-class="shadow-lg"",
    "no-caret=""",
    " ...",
    ">",
    "
",
    " at <TimezoneOffset",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "number" with v-bind value \`false\` will render number="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <BFormInput",
    "type="range"",
    "min="-12"",
    "max="14"",
    " ...",
    ">",
    "
",
    " at <BDropdown",
    "class="timezone-dropdown w-100"",
    "menu-class="shadow-lg"",
    "no-caret=""",
    " ...",
    ">",
    "
",
    " at <TimezoneOffset",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "plaintext" with v-bind value \`false\` will render plaintext="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <BFormInput",
    "type="range"",
    "min="-12"",
    "max="14"",
    " ...",
    ">",
    "
",
    " at <BDropdown",
    "class="timezone-dropdown w-100"",
    "menu-class="shadow-lg"",
    "no-caret=""",
    " ...",
    ">",
    "
",
    " at <TimezoneOffset",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "trim" with v-bind value \`false\` will render trim="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <BFormInput",
    "type="range"",
    "min="-12"",
    "max="14"",
    " ...",
    ">",
    "
",
    " at <BDropdown",
    "class="timezone-dropdown w-100"",
    "menu-class="shadow-lg"",
    "no-caret=""",
    " ...",
    ">",
    "
",
    " at <TimezoneOffset",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "noWheel" with v-bind value \`false\` will render noWheel="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <BFormInput",
    "type="range"",
    "min="-12"",
    "max="14"",
    " ...",
    ">",
    "
",
    " at <BDropdown",
    "class="timezone-dropdown w-100"",
    "menu-class="shadow-lg"",
    "no-caret=""",
    " ...",
    ">",
    "
",
    " at <TimezoneOffset",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) (2)",
    "
",
    " at <BDropdown",
    "class="timezone-dropdown w-100"",
    "menu-class="shadow-lg"",
    "no-caret=""",
    " ...",
    ">",
    "
",
    " at <TimezoneOffset",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "dropleft" with v-bind value \`false\` will render dropleft="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <BDropdown",
    "class="timezone-dropdown w-100"",
    "menu-class="shadow-lg"",
    "no-caret=""",
    " ...",
    ">",
    "
",
    " at <TimezoneOffset",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "dropright" with v-bind value \`false\` will render dropright="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <BDropdown",
    "class="timezone-dropdown w-100"",
    "menu-class="shadow-lg"",
    "no-caret=""",
    " ...",
    ">",
    "
",
    " at <TimezoneOffset",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "dropup" with v-bind value \`false\` will render dropup="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <BDropdown",
    "class="timezone-dropdown w-100"",
    "menu-class="shadow-lg"",
    "no-caret=""",
    " ...",
    ">",
    "
",
    " at <TimezoneOffset",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "noFlip" with v-bind value \`false\` will render noFlip="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <BDropdown",
    "class="timezone-dropdown w-100"",
    "menu-class="shadow-lg"",
    "no-caret=""",
    " ...",
    ">",
    "
",
    " at <TimezoneOffset",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "right" with v-bind value \`false\` will render right="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <BDropdown",
    "class="timezone-dropdown w-100"",
    "menu-class="shadow-lg"",
    "no-caret=""",
    " ...",
    ">",
    "
",
    " at <TimezoneOffset",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "block" with v-bind value \`false\` will render block="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <BDropdown",
    "class="timezone-dropdown w-100"",
    "menu-class="shadow-lg"",
    "no-caret=""",
    " ...",
    ">",
    "
",
    " at <TimezoneOffset",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) (2)",
    "
",
    " at <BDropdown",
    "class="timezone-dropdown w-100"",
    "menu-class="shadow-lg"",
    "no-caret=""",
    " ...",
    ">",
    "
",
    " at <TimezoneOffset",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "split" with v-bind value \`false\` will render split="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <BDropdown",
    "class="timezone-dropdown w-100"",
    "menu-class="shadow-lg"",
    "no-caret=""",
    " ...",
    ">",
    "
",
    " at <TimezoneOffset",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "active" with v-bind value \`false\` will render active="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <BLink",
    "href="https://www.timeanddate.com/time/zones/"",
    "target="_blank"",
    ">",
    "
",
    " at <TimezoneOffset",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) (3)",
    "
",
    " at <BLink",
    "href="https://www.timeanddate.com/time/zones/"",
    "target="_blank"",
    ">",
    "
",
    " at <TimezoneOffset",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "exact" with v-bind value \`false\` will render exact="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <BLink",
    "href="https://www.timeanddate.com/time/zones/"",
    "target="_blank"",
    ">",
    "
",
    " at <TimezoneOffset",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "exactPath" with v-bind value \`false\` will render exactPath="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <BLink",
    "href="https://www.timeanddate.com/time/zones/"",
    "target="_blank"",
    ">",
    "
",
    " at <TimezoneOffset",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "noPrefetch" with v-bind value \`false\` will render noPrefetch="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <BLink",
    "href="https://www.timeanddate.com/time/zones/"",
    "target="_blank"",
    ">",
    "
",
    " at <TimezoneOffset",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "replace" with v-bind value \`false\` will render replace="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <BLink",
    "href="https://www.timeanddate.com/time/zones/"",
    "target="_blank"",
    ">",
    "
",
    " at <TimezoneOffset",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
]
`;
